### PR TITLE
AnimationParser: add additional check

### PR DESCRIPTION
**Pull request by lemirep**
To make sure we have the right number of components for a given channel type

### CODE REVIEW

File: src/core/gltf2importer/animationparser.cpp [hunk 1]
```diff
@@ -173,6 +173,19 @@ QString channelPathToProperty(const QString &path)
...
+{
+    if (channelName == QStringLiteral("translation"))
+        return 3;
+    if (channelName == QStringLiteral("rotation"))
+        return 4;
+    if (channelName == QStringLiteral("scale"))
+        return 3;
+    if (channelName == QStringLiteral("weights"))
+        return 1;
+    return 0;
```
Comment (mkrus) [inline src/core/gltf2importer/animationparser.cpp]: might be good, if the result is 0, to print a warning that the channel name is invalid

File: src/core/gltf2importer/animationparser.cpp [hunk 2]
```diff
@@ -249,6 +262,14 @@ Qt3DAnimation::QChannel AnimationParser::animationChannelDataFromBuffer(const QS
     std::vector<float> valueStamps;
     std::tie(nbComponents, valueStamps) = animationChannelDataFromData(sampler);
 
+    // Check nbComponent and type are what is expected for a given channel
+    const quint8 expectedComponents = expectedComponentsCountForChannel(channelName);
+
```
Comment (mkrus) [inline src/core/gltf2importer/animationparser.cpp]: and if warning is printed above, have an early return on 0 result (not printing anything else)

File: src/core/gltf2importer/animationparser.cpp (hunk 1 above)
Comment (lemirep) [reply src/core/gltf2importer/animationparser.cpp]: done

File: src/core/gltf2importer/animationparser.cpp (hunk 2 above)
Comment (lemirep) [reply src/core/gltf2importer/animationparser.cpp]: done

**Pull request merged**